common/server_room.cpp

```diff
@@ -232,6 +234,22 @@ void Server_Room::say(const QString &userName, const QString &s, bool sendToIsl)
     event.set_name(userName.toStdString());
     event.set_message(s.toStdString());
     sendRoomEvent(prepareRoomEvent(event), sendToIsl);
+ 
+    if (chatHistorySize != 0) {
+
+        ServerInfo_ChatMessage chatMessage;
+        QDateTime dateTime = dateTime.currentDateTime();
+        QString dateTimeString = dateTime.toString("MM/dd/yy hh:mm:ss");
+        chatMessage.set_time(dateTimeString.toStdString());
+        chatMessage.set_sender_name(userName.toStdString());
+        chatMessage.set_message(s.simplified().toStdString());
+
+        if (chatHistory.size() > chatHistorySize)
```

Comment:
From my tests i always receive chatHistorySize+1 messages in the backlog (eg. if in the config i set chathistorysize=3, an user joining receives 4 messages). I think this is due to the fact that here you are using > instead of >=.